core/templates/dev/head/galleries/Gallery.js

```diff
@@ -99,8 +99,10 @@ oppia.factory('searchService', [
         _lastSelectedCategories = angular.copy(selectedCategories);
         _lastSelectedLanguageCodes = angular.copy(selectedLanguageCodes);
         _searchCursor = data.search_cursor;
+
         $rootScope.$broadcast(
           'refreshGalleryData', data, hasPageFinishedLoading());
+        $translate.refresh();
```

Comment:
Can you add a comment saying why this is necessary? (IIUC, it's the only place we do it.)